From c5c528beb6e1cfed3ec93a3a264084df32ce83c2 Mon Sep 17 00:00:00 2001
From: Hui Peng <phui@google.com>
Date: Thu, 4 Jan 2024 06:27:52 +0000
Subject: [PATCH] Fix a security bypass issue in
 access_secure_service_from_temp_bond

Bug: 318374503
Test: m com.android.btservices | manual test against PoC | QA
Ignore-AOSP-First: security
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:62944f39f502b28687a5142ec2d77585525591bc)
Merged-In: I48df2c2d77810077e97d4131540277273d441998
Change-Id: I48df2c2d77810077e97d4131540277273d441998
---
 system/stack/btm/btm_sec.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/system/stack/btm/btm_sec.cc b/system/stack/btm/btm_sec.cc
index c77c40535f..6ec6d16297 100644
--- a/system/stack/btm/btm_sec.cc
+++ b/system/stack/btm/btm_sec.cc
@@ -222,8 +222,7 @@ static bool access_secure_service_from_temp_bond(const tBTM_SEC_DEV_REC* p_dev_r
                                                  bool locally_initiated,
                                                  uint16_t security_req) {
   return !locally_initiated && (security_req & BTM_SEC_IN_AUTHENTICATE) &&
-    p_dev_rec->is_device_authenticated() &&
-    p_dev_rec->is_bond_type_temporary();
+         p_dev_rec->is_bond_type_temporary();
 }
 
 /*******************************************************************************
-- 
2.43.0.594.gd9cf4e227d-goog

